Packet++/src/IPv6Layer.cpp

```diff
@@ -33,6 +33,12 @@ IPv6Layer::IPv6Layer(uint8_t* data, size_t dataLen, Layer* prevLayer, Packet* pa
 	m_LastExtension = NULL;
 	m_ExtensionsLen = 0;
 
+	if (dataLen < sizeof(ip6_hdr))
```

Comment:
This is not needed if before getting to the c'tor we call `IPv6Layer::isDataValid(...)`